runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/ExecutionStateTracker.java

```diff
@@ -335,6 +352,19 @@ protected void takeSampleOnce(long millisSinceLastSample) {
       transitionsAtLastSample = transitionsAtThisSample;
     }
     updateMillisSinceLastTransition(millisSinceLastSample, state);
+    updateMillisSinceBundleStart(millisSinceLastSample);
+  }
+
+  // Override this to implement bundle level lull reporting.
+  protected void reportBundleLull(Thread trackedThread, long millisSinceBundleStart) {}
+
+  @SuppressWarnings("NonAtomicVolatileUpdate")
+  private void updateMillisSinceBundleStart(long millisSinceLastSample) {
+    millisSinceBundleStart += millisSinceLastSample;
+    if (millisSinceBundleStart > nextBundleLullReportMs) {
+      reportBundleLull(trackedThread, millisSinceBundleStart);
+      nextBundleLullReportMs += BUNDLE_LULL_REPORT_MS;
+    }
   }
 
```

Comment:
maybe a BEAM/Java developer can help here.  Is there a reason we supress these warnings?  Is it safe to do for some reason?  It seems to indicate a possible data race which we should fix?

Reply:
Add a comment that the ignore warning is bc. updateMillisSinceBundleStart will only ever be called by the thread doing the dofn processing.  Reset is where multiple threads may call and there they are doing blind writes.